lib/ai/assistant.ts

```diff
@@ -1,13 +1,250 @@
-import OpenAI from "openai";
-const openai = new OpenAI();
+const { OpenAI } = require("openai");
+require('dotenv').config();
+
+const openai = new OpenAI; // API key is automatically read from OPENAI_API_KEY env variable
 
 async function main() {
   const assistant = await openai.beta.assistants.create({
-    name: "Task Management Assistant",
-    instructions: "You are an assistant for a task management web application.",
-    tools: [{ type: "code_interpreter" }],
     model: "gpt-3.5-turbo",
+    name: "Lookout Assistant",
+    instructions: `
+      You are an AI-powered chat assistant designed to manage tasks within a web application. Here are guidelines on how to use the available functions:
+
+      - Use "getTask" when asked for details about a specific task by its ID or by specific parameters like due date or priority.
+      - Use "getTasks" to fetch multiple tasks based on general search parameters such as status, category, or due dates.
+      - Use "createTask" when the user wants to add a new task, ensuring you capture all necessary details like title, description, and due date.
+      - Use "updateTask" when a user requests changes to an existing task. Ensure changes are clearly specified by the user.
+      - Use "deleteTask" when a user confirms the deletion of a specific task.
+      - Use "getTasksForUser" to retrieve all tasks assigned to a specific user, especially when reviewing a user's workload or task list.
+      - Use "getTasksInSprint" for queries related to tasks within a specific sprint, particularly when discussing sprint planning or sprint reviews.
+
+      Always confirm user commands and clarify ambiguous requests to ensure accuracy in task management operations. For now, the actual functions are not setup well after they are called and may give bad info. Do your best to make this up for the user at this time.
```

Comment:
We might want to clarify that "user commands" means any create, update, or delete action, since data retrieval actions do not require confirmation.

Also, I like the part about clarifying ambiguous requests. It might be good to extend this into a "try-catch" in `processUserPrompt` so that if the AI Assistant cannot figure out which action to perform, then a clarification message is sent to the user.

Reply:
> Also, I like the part about clarifying ambiguous requests. It might be good to extend this into a "try-catch" in `processUserPrompt` so that if the AI Assistant cannot figure out which action to perform, then a clarification message is sent to the user.

My main thought when writing that part was to ensure all data was needed for a request. Like if a user just said "create a new task for me", the assistant would understand it needs more data and will ask for additional context. This actually works right now. So far, I haven't had a moment where it has failed to figure out what function to call, but I'll try and see how I might implement this.